src/openforms/forms/migrations/0042_explanation_template_data_migration.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 3.2.14 on 2022-08-26 15:29
+from django.db import migrations
+
+
+def append_explanation_template_message(apps, schema_editor):
+    from openforms.authentication.registry import register as auth_register
+    from django.utils.translation import gettext_lazy as _
+
+    Form = apps.get_model("forms", "Form")
+    for form in Form.objects.all():
+        plugins = form.authentication_backends
+        login_options_count = len(
+            [plugin_id for plugin_id in plugins if plugin_id in auth_register._registry]
+        )
+        can_login = login_options_count > 0
+        form_login_required = any(
+            form_step.form_definition.login_required
+            for form_step in form.formstep_set.all()
+        )
+
+        if form_login_required:
+            message = _("Please authenticate to start the form.")
+        elif can_login:
+            message = _("Please authenticate or start the form anonymously.")
+        else:
+            message = _("Please click the button below to start the form.")
+
+        form.explanation_template += f'<p id="start-form">{message}</p>'
```

Comment:
if there is HTML in the translated messages, you now have broken HTML/possible script injection.

Use `django.utils.html.format_html` when building HTML in python.

Reply:
Fortunately there is no html in the translated messages. Good tip :)

Reply:
Yeah until sometimes slips by and another contributor adds HTML in the translation messages.

Security should always be in your mind whenever you're dealing with HTML outside of the template engine. It's not a tip, it's pointing out a security vulnerability.

Reply:
> Can you provide me a source?

You really should be able to see what happens if `message` contains HTML tags (possibly `<script>` tags). They will not be escaped.

For a source, I'm fairly certain that XSS is in the OWASP top 10.

> What would happen if a contributor added HTML in the translation message?

See above, it will not be escaped.

Reply:
@dariomory Please use  `django.utils.html.format_html` and don't use the `id="start-form"` in it because it will be lost once someone saves the form again in the admin.